packages/flutter_tools/test/application_package_test.dart

```diff
@@ -26,13 +26,13 @@ final Map<Type, Generator> noColorTerminalOverride = <Type, Generator>{
 
 void main() {
   group('ApkManifestData', () {
-    test('Select explicity enabled activity', () {
+    test('Select explicity enabled, launcher and main activity', () {
```

Comment:
I'm still not sure what these test names mean.  What is the specific action that they are testing, and what is the result that they are verifying?

Reply:
Prior logic just check whether an activity is configured enabled, however, the right logic should be that:
Launcher activity should be 
```
 <action android:name="android.intent.action.MAIN" />
 <category android:name="android.intent.category.LAUNCHER" />
```
and enabled(it will be true if not specified).

Those tests are designed to check this logic.

Reply:
Ok, it looks like these tests should probably be named:

'Parses manifest with an Activity that has enabled set to true', and

'Parses manifest with an Activity that has no value for its enabled field'

Do those names correctly capture the purpose of these tests?

Reply:
@matthew-carroll 
I've updated the test description and it would be better now.